app/prepare/protection.py

```diff
@@ -0,0 +1,24 @@
+class Protection:
+    def __init__(self, knight: dict):
+        self.knight = knight
+        self.knight["protection"] = 0
+
+    def get_protection(self):
+        for a in self.knight["armour"]:
+            self.knight["protection"] += a["protection"]
+
+    def get_weapon(self):
+        self.knight["power"] += self.knight["weapon"]["power"]
+
+    def get_potion(self):
+        if self.knight["potion"] is not None:
+            if "power" in self.knight["potion"]["effect"]:
+                self.knight["power"] +=\
+                    self.knight["potion"]["effect"]["power"]
+
+            if "protection" in self.knight["potion"]["effect"]:
+                self.knight["protection"] +=\
+                    self.knight["potion"]["effect"]["protection"]
+
+            if "hp" in self.knight["potion"]["effect"]:
+                self.knight["hp"] += self.knight["potion"]["effect"]["hp"]
```

Comment:
These code duplications can be fixed, using `stats = ("protection", "power", "hp")` and looping on `stats`

Reply:
oh. i will do it